config/drupal-10/drupal-10.2-deprecations.php

```diff
@@ -24,4 +25,9 @@
         new MethodToMethodWithCheckConfiguration('Drupal\system\Plugin\ImageToolkit\GDToolkit', 'getResource', 'getImage'),
         new MethodToMethodWithCheckConfiguration('Drupal\system\Plugin\ImageToolkit\GDToolkit', 'setResource', 'setImage'),
     ]);
+
+    // https://www.drupal.org/node/3401941
+    $rectorConfig->ruleWithConfiguration(RenameClassRector::class, [
+        'Drupal\\Tests\\field\\Traits\\EntityReferenceTestTrait' => 'Drupal\\Tests\\field\\Traits\\EntityReferenceFieldCreationTrait',
+    ]);
```

Comment:
@bbrala should we wrap this with a BC protection? I guess then we'd need to extend `RenameClassRector` _and_ a config object to do so :/ or is there an easier way?

Reply:
Yeah we need to wrap. I've wanted to wrap for a few times now, but haven't taken the time to see if its ewasily possible.

You'd need to do quite a few weird things to make sure things are instantiated correctly. Would open up loads of easy things like this rector though.

Reply:
Since all classes in rector are final, there is no extending.

Reply:
I believe this PR is held up for the BC refactor?

Reply:
Ok, visited this again. Something generic is very hard. Something expanding on what RenameClassRector does AND hooking into the file is kinda hard. 

So guess it needs a specific rector for this change, which is sad. And my time is up for today ;x